### PR TITLE
BO password reset with maildev

### PR DESCRIPTION
**Pull request by SD1982**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop 
| Description?      | New test for the password reminder in BO
| Type?             | refacto 
| Category?         | TE 
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | ##
| How to test?      | Can be tested with this command : `TEST_PATH="functional/BO/00_login/01_passwordReminder" URL_FO="your FO url" npm run specific-test`
| Possible impacts? | nothing

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/23543)

### CODE REVIEW

File: tests/UI/pages/BO/login/index.js [hunk 1]
```diff
@@ -47,6 +52,28 @@ class Login extends BOBasePage {
...
+  }
+
+  /**
+   * Get and return reset password success message text
+   * @param page
+   * @returns {Promise<*>}
+   */
+  async checkSendResetPasswordLinkSuccess(page) {
+    return this.getTextContent(page, this.resetPasswordSuccessConfirmationText);
+  }
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/login/index.js]:
```suggestion
  /**
   * Go to password reset page and send reset password link
   * @param page
   * @param employeeEmail
   * @returns {Promise<void>}
   */
  async sendResetPasswordLink(page, employeeEmail) {
    await page.click(this.forgotPasswordLink);
    await this.waitForVisibleSelector(page, this.resetPasswordButton);
    await this.setValue(page, this.resetPasswordEmailFormField, employeeEmail);
    await page.click(this.resetPasswordButton);
  }

  /**
   * Get and return reset password success message text
   * @param page
   * @returns {Promise<string>}
   */
  getResetPasswordSuccessMessage(page) {
    return this.getTextContent(page, this.resetPasswordSuccessConfirmationText);
  }
```

File: tests/UI/campaigns/functional/BO/00_login/01_passwordReminder.js [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+
+      const pageTitle = await loginPage.getPageTitle(page);
+      await expect(pageTitle).to.contains(loginPage.pageTitle);
+    });
+
+    it('should send reset password mail', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'sendResetPasswordMailAndCheckSuccess', baseContext);
+      await loginPage.sendBOResetPasswordLink(page, createEmployeeData.email);
+
+      const successTextContent = await loginPage.checkSendResetPasswordLinkSuccess(page);
```
Comment (boubkerbribri) [inline tests/UI/campaigns/functional/BO/00_login/01_passwordReminder.js]:
```suggestion
      const successTextContent = await loginPage.getResetPasswordSuccessMessage(page);
```

File: tests/UI/campaigns/functional/BO/00_login/01_passwordReminder.js [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+      await testContext.addContextItem(this, 'testIdentifier', 'goToBOLoginPage', baseContext);
+      await loginPage.goTo(page, global.BO.URL);
+
+      const pageTitle = await loginPage.getPageTitle(page);
+      await expect(pageTitle).to.contains(loginPage.pageTitle);
+    });
+
+    it('should send reset password mail', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'sendResetPasswordMailAndCheckSuccess', baseContext);
+      await loginPage.sendBOResetPasswordLink(page, createEmployeeData.email);
```
Comment (boubkerbribri) [inline tests/UI/campaigns/functional/BO/00_login/01_passwordReminder.js]:
```suggestion
      await loginPage.sendResetPasswordLink(page, createEmployeeData.email);
```

File: tests/UI/campaigns/functional/BO/00_login/01_passwordReminder.js [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+
+const baseContext = 'functional_BO_login_passwordReminder';
+
+let browserContext;
+let page;
+let numberOfEmployees = 0;
+// maildev
+let newMail;
+const {smtpServer, smtpPort} = global.maildevConfig;
+const resetPasswordSuccessText = 'Please, check your mailbox.';
```
Comment (boubkerbribri) [inline tests/UI/campaigns/functional/BO/00_login/01_passwordReminder.js]:
```suggestion
```

this message should be added to the page

File: tests/UI/campaigns/functional/BO/00_login/01_passwordReminder.js [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+// Import test context
+const testContext = require('@utils/testContext');
+
+const baseContext = 'functional_BO_login_passwordReminder';
+
+let browserContext;
+let page;
+let numberOfEmployees = 0;
+// maildev
+let newMail;
```
Comment (boubkerbribri) [inline tests/UI/campaigns/functional/BO/00_login/01_passwordReminder.js]:
```suggestion

// Maildev config and vars
let newMail;
```

File: tests/UI/campaigns/functional/BO/00_login/01_passwordReminder.js [hunk 6]
```diff
@@ -0,0 +1,273 @@
...
+    it('should go to "Advanced parameters>Team" page', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'goToAdvancedParamsPage', baseContext);
+
+      await dashboardPage.goToSubMenu(
+        page,
+        dashboardPage.advancedParametersLink,
+        dashboardPage.teamLink,
+      );
+
+      await employeesPage.closeSfToolBar(page);
```
Comment (boubkerbribri) [inline tests/UI/campaigns/functional/BO/00_login/01_passwordReminder.js]:
```suggestion
```

Could be deleted, only needed one time after login

File: tests/UI/campaigns/functional/BO/00_login/01_passwordReminder.js [hunk 7]
```diff
@@ -0,0 +1,273 @@
...
+      const textResult = await emailPage.sendTestEmail(page, global.BO.EMAIL);
+      await expect(textResult).to.contains(emailPage.sendTestEmailSuccessfulMessage);
+    });
+
+    it('should check if test mail is received', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'checkMailBox', baseContext);
+
+      await expect(newMail.subject).to.contains(testMailSubject);
+    });
+
```
Comment (boubkerbribri) [inline tests/UI/campaigns/functional/BO/00_login/01_passwordReminder.js]:
```suggestion
```

We should not test this in this scenario, you can delete it and later we will crate the proper scenario for this.

File: tests/UI/campaigns/functional/BO/00_login/01_passwordReminder.js [hunk 8]
```diff
@@ -0,0 +1,273 @@
...
+// init mailListener
+const mailListener = mailHelper.createMailListener();
+
+describe('BO Password reminder', async () => {
+  // before and after functions
+  before(async function () {
+    browserContext = await helper.createBrowserContext(this.browser);
+    page = await helper.newTab(browserContext);
+    mailHelper.startListener(mailListener);
```
Comment (boubkerbribri) [inline tests/UI/campaigns/functional/BO/00_login/01_passwordReminder.js]:
```suggestion
    browserContext = await helper.createBrowserContext(this.browser);
    page = await helper.newTab(browserContext);
    
    // Start listening to maidev server
    mailHelper.startListener(mailListener);
```

Just to improve readability

File: tests/UI/campaigns/functional/BO/00_login/01_passwordReminder.js [hunk 9]
```diff
@@ -0,0 +1,273 @@
...
+    // Handle every new email
+    mailListener.on('new', (email) => {
+      newMail = email;
+    });
+  });
+
+  after(async () => {
+    await helper.closeBrowserContext(browserContext);
+    mailHelper.stopListener(mailListener);
```
Comment (boubkerbribri) [inline tests/UI/campaigns/functional/BO/00_login/01_passwordReminder.js]:
```suggestion
    await helper.closeBrowserContext(browserContext);
    
    // Stop listening to maidev server
    mailHelper.stopListener(mailListener);
```

Just to improve readability

File: tests/UI/campaigns/functional/BO/00_login/01_passwordReminder.js [hunk 10]
```diff
@@ -0,0 +1,273 @@
...
+    });
+
+    it('should create employee and check result', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'createEmployee', baseContext);
+
+      const textResult = await addEmployeePage.createEditEmployee(page, createEmployeeData);
+      await expect(textResult).to.equal(employeesPage.successfulCreationMessage);
+
+      const numberOfEmployeesAfterCreation = await employeesPage.getNumberOfElementInGrid(page);
+      await expect(numberOfEmployeesAfterCreation).to.be.equal(numberOfEmployees + 1);
```
Comment (boubkerbribri) [inline tests/UI/campaigns/functional/BO/00_login/01_passwordReminder.js]:
```suggestion
      await expect(textResult).to.equal(employeesPage.successfulCreationMessage);
```

No need for this check here (to part of the test)

File: tests/UI/campaigns/functional/BO/00_login/01_passwordReminder.js [hunk 11]
```diff
@@ -0,0 +1,273 @@
...
+      await loginCommon.logoutBO(this, page);
+    });
+  });
+
+  describe('Go to BO and reset to default mail parameters', async () => {
+    it('should login in BO', async function () {
+      await loginCommon.loginBO(this, page);
+    });
+
+    it('should go to email setup page', async function () {
```
Comment (boubkerbribri) [inline tests/UI/campaigns/functional/BO/00_login/01_passwordReminder.js]:
```suggestion
  });

  describe('Go to BO and reset to default mail parameters', async () => {
    it('should go to email setup page', async function () {
```

We could avoid logout and login Here

**Review by boubkerbribri — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Comment by Progi1984** [comment]
Thanks @SD1982

**Pull request merged**